### DISCUSSION TITLE
Suddenly, login with my 2FA doesn't work anymore.

### DISCUSSION THREAD

**Discussion by Potoman** [community/community#140103] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi,

I'm trying to connect to my account after a while (maybe 1 or 2 month), but the 2FA step through TOTP application don't work. The 6 digit I write in Github login page doesn't work at all.

I received an email from github which say :

```
Hey Potoman!

Are you trying to sign in? Based on your typical account activity, we did not recognize the location of the sign in. As of this email, two-factor authentication has prevented the sign in attempt from being successful. Your password was entered correctly, but an authentication code was not entered correctly. You may view details about this sign in attempt by visiting ...
``` 

So I understand that my password is ok.

I successfully connect to my account by using a recovery code.

I try to use the support page : If you run into problems, please contact support by visiting "https://github.com/contact"

But it is seriously a bad interface. The "AI" always asking me my username which is "Potoman" and don't recognize it.

I realy don't like 2FA but I understand the necessity.

If Github, as he say in its emaill, doesn't recognize my location, how can I get rid of it !!!!!!!!?????? So I can be stuck because an algorithm is not happy with my current location ? Is it serious ? I really find this things dangerous for my repository and I am really afraid to lose my connection at all. This doesn't give me confidence in Github.

To be honnest, if I traval all over the world I expect to be able to log into my account.

What things I can do to see my 2FA working again ?

Thanks for you attention.

++

**Comment by AidenJamesYt** [comment]
Really the only thing you can do is:

1. try to recover the 2fa account/key
2. try to find 2fa backup codes that you saved
3. contact support

**Reply by Potoman** [reply]
What do you mean by your point 1 ?
Does I have to reset the TOTP application ?
Does I have to re-do the 2FA setup step from Github procedure ?

Thx.

**Comment by Potoman** [comment, marked as answer]
For those who have had this append, I found a solution.

So for the 2FA application on my cellphone, I used TOTP. In this app, I remove all of my current credential. On my Github profile (I log through a recovery code), I generate a new QR code. I scan it on TOTP but the given code still doesn't work.
I download [2FAS](https://play.google.com/store/apps/details?id=com.twofasapp&hl=fr&pli=1) app. Scan the same QR code, tap the 6 characters code and all work.
So... I don't explain why TOTP doesn't work. I check the update I have installed on my Cellphone but my Android tell me the app is up to date so...

Hope this solution help someone.

++

**Reply by MasterInQuestion** [reply]
Time matters for TOTP (Time-based One-Time Password).
    GitHub asserts which under 30 s:
    https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication#manually-configuring-a-totp-app

    It also much depends on the quality of the TOTP app implementation:
    Some may not compensate the system time zone differences. (thus fail)

    See also:
    https://github.com/orgs/community/discussions/147499
